### ISSUE TITLE
Rule to check users don't accidentally specify CallerArgumentExpression parameters

### ISSUE THREAD

**Issue by manfred-brands** [nunit/nunit.analyzers#639]
To improve messages for failing tests, NUnit4 uses `CallerMemberExpresion` to convert the _actual_ and _constraint_ parameters into text.

Users used to the old format spec, might accidentally pass in two paramters without realizing that this doesn't work.

```csharp
Assert.That(actual, Is.EqualTo(expected), "Expected {0}, but got {1}", expected, actual);
```

In NUnit 4, this should have been specified as:

```csharp
Assert.That(actual, Is.EqualTo(expected), $"Expected {expected}, but got {actual}");
```
